http/http_client.h

```diff
@@ -0,0 +1,432 @@
+/* http_client.h                                                   -*- C++ -*-
+   This file is part of MLDB.
+
+   Wolfgang Sourdeau, January 2014
+   Copyright (c) 2014 Datacratic.  All rights reserved.
+
+   An asynchronous HTTP client.
+
+   HttpClient is meant to provide a featureful, generic and asynchronous HTTP
+   client class. It supports strictly asynchronous (non-blocking) operations,
+   HTTP pipelining and concurrent requests while enabling streaming responses
+   via a callback mechanism. It is meant to be subclassed whenever a
+   synchronous interface or a one-shot response mechanism is required. In
+   general, the code should be complete enough that existing and similar
+   classes could be subclassed gradually (HttpRestProxy, s3 internals). As a
+   generic class, it does not make assumptions on the transferred contents.
+   Finally, it is based on the interface of HttpRestProxy.
+
+   Caveat:
+   - since those require header interpretation, there is not support for
+     cookies per se
+*/
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "mldb/ext/jsoncpp/value.h"
```

Comment:
Would be good to not include the async_event_source header here, and to put more of the methods in the .cc file.